one_shot_scripts/add_blog_names.js

```diff
@@ -7,22 +7,52 @@ const {
 	generate_short_id,
 } = require('../serverjs/cubefn.js');
 
+const batch_size = 100;
+
+const cubeNameCache = {};
+const userNameCache = {};
+
+async function addVars(blog) {
+    if(!cubeNameCache[blog.cube]) {
+        const cube = await Cube.findById(blog.cube);
+        cubeNameCache[blog.cube] = cube ? cube.name : 'Cube';
+    }
+    blog.cubename = cubeNameCache[blog.cube];
+
+    if(!userNameCache[blog.owner]) {
+        const user = await User.findById(blog.owner);
+        userNameCache[blog.owner] = user ? user.username : 'User';
+    }
+    blog.username = userNameCache[blog.owner];
+
+    return await blog.save();
+}
+
 (async () => {
     var i = 0;
 	mongoose.connect(mongosecrets.connectionString).then( async (db) => {
-        const user = await User.findById('5cf9cdf2aefc6508c5ebcb40');
-    
-        console.log(user);
-    
-        const cursor =  Blog.find({owner: {$in: user.followed_users}}).cursor();
-        for (let blog = await cursor.next(); blog != null; blog = await cursor.next()) {
-            console.log(i++);
-            const cube = await Cube.findById(blog.cube);
-            const user = await User.findById(blog.owner);
-            blog.cubename = cube ? cube.name : 'Cube';
-            blog.username = user ? user.username : 'User';
-            await blog.save();
+
+        const count = await Blog.countDocuments();
+        const cursor = Blog.find().cursor();
+        
+        //batch them in 100
+        for(var i = 0; i < count; i += batch_size)
+        {
+            console.log("Finished: " + i + " of " + count + " blogs");
+            const blogs = [];
+            for(var j = 0; j < batch_size; j++)
+            {
+                if(i + j < count)
+                {
+                    let blog = await cursor.next()
+                    if(blog) {
+                        blogs.push(blog)
+                    }
+                }
+            }
+            await Promise.all(blogs.map(blog => addVars(blog)));
         }
+        console.log("Finished: " + count + " of " + count + " blogs");
```

Comment:
This log is out the for, the one in add_deck_names is inside, which do you prefer?

Reply:
Pretty sure you want inside?